### ISSUE TITLE
What's the output shape of gather

### ISSUE THREAD

**Issue by mzmssg** [onnx/onnx#2765]
I expected gather works like `numpy.take` because it‘s used to verify gather result.
But in [Gather doc](https://github.com/onnx/onnx/blob/master/docs/Operators.md#gather)
```
  data = [
      [1.0, 1.2, 1.9],
      [2.3, 3.4, 3.9],
      [4.5, 5.7, 5.9],
  ]
  indices = [
      [0, 2],
  ]
  axis = 1,
  output = [
      [
          [1.0, 1.9],
          [2.3, 3.9],
          [4.5, 5.9],
      ],
  ] 
# output_shape (1, 3, 2)
```
Numpy result is
```
import numpy as np

data = [
    [1.0, 1.2, 1.9],
    [2.3, 3.4, 3.9],
    [4.5, 5.7, 5.9],
]
indices = [
    [0, 2],
]
out = np.take(data, indices, axis=1)
print(out.shape) 
# (3, 1, 2)
```
Literally, it seems to be (1, 2, 3) (i_0, i_1 from indices and j_0 from data)
> Let k = indices[i_{0}, ..., i_{q-1}] Then output[i_{0}, ..., i_{q-1}, j_{0}, ..., j_{r-2}] = input[j_{0}, k, j_{1}, ..., j_{r-2}]

So which is the expect output shape?

**Comment by TMVector** [comment]
Hi @mzmssg, thanks for the report. The [code examples](https://github.com/onnx/onnx/blob/master/docs/Operators.md#examples-44) in the specification should be checked against ONNX backends, so they should be correct. The code you quote from the operator description could well be wrong (it looks like it is) -- feel free to raise a PR fixing the description. You would need to edit the description string at [onnx/defs/tensor/defs.cc#L1068](https://github.com/onnx/onnx/blob/master/onnx/defs/tensor/defs.cc#L1068) string and regenerate the docs with [tools/update_doc.sh](https://github.com/onnx/onnx/blob/master/tools/update_doc.sh)

**Comment by mzmssg** [comment]
@TMVector Get you, will fix the doc later

**Comment by CyanHillFox** [comment]
@TMVector @mzmssg The document is still not correct.

**Comment by daquexian** [comment]
@mzmssg @TMVector @CyanHillFox Fixed by #3040
